ui/lib/CreateStream.tsx

```diff
@@ -188,21 +188,37 @@ const CreateStream = (props: CreateStreamProps) => {
   });
 
   // random string for the demo;
-  const renderReasonCode = () => {
+  const renderReasonCode = async () => {
     return Math.random().toString(36).substring(7);
   };
 
-  const createStream = async () => {
-    // Trigger function to get reason string (to be updated with props.renderReasonCode)
-    const reasonString = renderReasonCode();
-    // Send and wait for stream open to complete
-    const stream = await sm.create(
-      props.tokenAddress,
-      props.amountPerSecond,
-      reasonString
-    );
-    // then update props.registerStream
-    props.registerStream(stream);
+  const [isProcessing, setIsProcessing] = useState<boolean>(false);
+  const [processTxError, setProcessTxError] = useState(null);
+  const [isProcessed, setIsProcessed] = useState<boolean>(false);
+
+  const createStream = () => {
+    setIsProcessing(true);
+
+    renderReasonCode()
+      .then((reasonString) => {
+        return sm
+          .create(props.tokenAddress, props.amountPerSecond, reasonString)
+          .then((result) => {
+            props.registerStream(result);
+            setIsProcessing(false);
+            setIsProcessed(true);
```

Comment:
wonder if it would make sense to move these display variables to the demo app, inside of `props.registerStream`